vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -651,3 +657,25 @@ func (s *GenericAPIServer) Eventf(eventType, reason, messageFmt string, args ...
 		klog.Warningf("failed to create event %s/%s: %v", e.Namespace, e.Name, err)
 	}
 }
+
+// terminationLoggingListener wraps the given listener and logs new connections
+// after the stopCh has been closed, i.e. when termination has begun.
+type terminationLoggingListener struct {
+	net.Listener
+	stopCh <-chan struct{}
+}
+
+func (l *terminationLoggingListener) Accept() (net.Conn, error) {
+	c, err := l.Listener.Accept()
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case <-l.stopCh:
+		klog.V(2).Infof("Accepted new connection from %s during termination", c.RemoteAddr())
```

Comment:
this only logs the new connections, right?
already established ones won't be logged.

Reply:
yep